### DISCUSSION TITLE
How to disabled pre-compilation

### DISCUSSION THREAD

**Discussion by motea927** [nuxt-modules/i18n#1782] [Q&A]
When production, I got error `The message format compilation is not supported in this build. Because message compiler isn't included. You need to pre-compilation all message format.`,
but I don't know how to disabled pre-compilation,
Seems this option can not work.

```ts
import VueI18nPlugin from '@intlify/unplugin-vue-i18n';
export default defineNuxtConfig({
 vite: {
    plugins: [
      VueI18nPlugin.vite({
        runtimeOnly: false,
        /* options */
      }),
    ],
  },
})
```

**Comment by motea927** [comment]
I also try this, but still get error
```ts
import VueI18nPlugin from '@intlify/unplugin-vue-i18n';
export default defineNuxtConfig({
 vite: {
       resolve: {
      alias: {
        'vue-i18n': 'vue-i18n/dist/vue-i18n.esm-bundler.js',
      },
    },
  },
})
```

**Comment by ineshbose** [comment]
Would you be able to share more context about your i18n usage (or better, if you can provide a repoduction) please?

**Reply by larsalbrecht** [reply]
We have the same issue like @motea927:
In a plugin, we fetch translations (json via http) and call then "nuxtApp.$i18n.mergeLocaleMessage(locale, \<translations\>)". Nothing special at all. 

In dev mode, all works fine. But after build, it does not work anymore.

Tried:
nuxt.config.ts
```
  alias: {
    'vue-i18n': 'vue-i18n/dist/vue-i18n.esm-bundler.js',
  },
```

nuxt.config.ts
```
  vite: {
    resolve: {
      alias: {
        'vue-i18n': 'vue-i18n/dist/vue-i18n.esm-bundler.js',
      },
    },
  },
```

nuxt.config.ts
```
  i18n: {
    vueI18nLoader: true,
    //...
    vueI18n: {
        runtimeOnly: false,
    },
  },
```

Our Plugin (shortened):
```
export default defineNuxtPlugin(async (nuxtApp) => {
  const locale = nuxtApp.$i18n.locale || nuxtApp.$i18n.defaultLocale

  if (process.server) {
    await nuxtApp.$fetchTranslations(locale.value)
  }
  nuxtApp.$i18n.mergeLocaleMessage(locale.value, nuxtApp.$translations.value[locale.value])
})
```

**Reply by ineshbose** [reply]
Hi, thanks for providing some more context - I'm just confused about where `vue-i18n` comes in; perhaps it's out of my knowledge. /cc @kazupon

**Reply by motea927** [reply]
@ineshbose I apologize for the late response,

Because `nuxt-modules/i18n` use `vue-i18n`,

I refer to [Vue-i18n](https://vue-i18n.intlify.dev/guide/advanced/optimization.html) and [vite-plugin-vue-i18n](https://github.com/intlify/bundle-tools/tree/main/packages/vite-plugin-vue-i18n#%EF%B8%8F-notice),

I understand that I need use `vue-i18n.esm-bundler.js`, but seems currently can't make it very easy.

**Comment by motea927** [comment]
Hi @larsalbrecht, We have similar situation, I need dynamic fetch translations,

finally I use [patch-package](https://www.npmjs.com/package/patch-package) to patch `node_modules/@nuxtjs/i18n/dist/runtime/plugin.mjs`'s soruce code,

I replace `import { createI18n } from "vue-i18n";` to

`import { createI18n } from "../../../../vue-i18n/dist/vue-i18n.esm-bundler";`,

Maybe it's not a good solution, but it can work, you can try it.

**Comment by kazupon** [comment]
In production, it's recommended that all resources be precompiled if possible.

The CLI to precompile resouces can be found here:
https://github.com/intlify/cli

And here is an example using dynamic loading in conjunction with back-end
https://github.com/intlify/vue-i18n-next/tree/master/examples/backend

These may be helpful for you

**Reply by kazupon** [reply]
If message-compiler is absolutely necessary for a production environment, we would consider adding it as an option, although we do not recommend it because the bundle size is lager.

**Comment by JBustin** [comment]
Hello, 

About this topic, is there any doc about the pre compilation purpose and configuration (except https://v8.i18n.nuxtjs.org/options/misc/#precompile) ?

I'm asking because I'm trying to migrate from nuxt2 to nuxt3, I'm using @nuxtjs/i18n@edge, and I'm a little bit surprise that a precompile call is done to the server.
I'm using a CDN, my nuxt server is never called directly, the POST call for pre compile is a real problem.
Also, my translation files are huge, and there is not placeholder inside.

So is it possible to disable it ? And what is the potential lost ?

Regards
